Wrappers/Python/cil/framework/framework.py

```diff
@@ -3645,7 +3645,7 @@ def __init__(self,
         else:
             raise TypeError('array must be a CIL type DataContainer or numpy.ndarray got {}'.format(type(array)))
 
-        if array.shape != geometry.shape:
+        if numpy.squeeze(array).shape != geometry.shape:
             raise ValueError('Shape mismatch {} {}'.format(array.shape, geometry.shape))
```

Comment:
If the array is (1, X, Y) can geometry.shape be (1, X, Y)? If that's possible the error message would be confusing 'ValueError: Shape mismatch (1, X, Y) and (1, X, Y)'. Maybe print the squeezed shape? Or squeeze the geometry for the test too?

Reply:
I modified the code to leave the base `DataContainer` as it was, and only apply the changes to the derived classes that have a geometry shape (and are then guaranteed) to not have shape with singletons dimensions.